libraries/entities/src/ShapeEntityItem.cpp

```diff
@@ -160,10 +160,22 @@ void ShapeEntityItem::appendSubclassData(OctreePacketData* packetData, EncodeBit
     APPEND_ENTITY_PROPERTY(PROP_ALPHA, getAlpha());
 }
 
-// This value specifes how the shape should be treated by physics calculations.  
-// For now, all polys will act as spheres
+// returns the "collision shape" for physics simulation
 ShapeType ShapeEntityItem::getShapeType() const {
-    return (_shape == entity::Shape::Cube) ? SHAPE_TYPE_BOX : SHAPE_TYPE_ELLIPSOID;
+    if (_shape == entity::Shape::Cube) {
+        return SHAPE_TYPE_BOX;
+    }
+    if (_shape == entity::Shape::Sphere) {
+        // SHAPE_TYPE_SPHERE means: "collide like true implicit sphere" whereas
+        // "SHAPE_TYPE_ELLIPSOID" means: "if spherical then use implicit sphere, otherwise use convex hull"
+        return SHAPE_TYPE_ELLIPSOID;
+    }
+    if (_shape == entity::Shape::Cylinder) {
+        return SHAPE_TYPE_CYLINDER_Y;
+    }
```

Comment:
This is only partially correct.  A `ShapeEntityItem` with `"shape" = "Cylinder"` could be given an asymmetrical `"dimensions"` such that its trivial cross-section is an ellipse rather than a circle.  However Bullet does not support such an implicit shape.  The only way to get close would be to use a `btConvexHullShape` with some optimized selection of points on the deformed cylinder's surface.  And note, if you used more than 42 points for the hull then our `ShapeFactory` would cull it down to only 42 (see the implementation of `createConvexHull()`).

But this is ok for this context.  Handling the deformed case should be done in `ShapeFactory.cpp`, similar to how the `SHAPE_TYPE_ELLIPSOID` case splits into two paths: the symmetrical (`btSphereShape`) and the asymmetrical (`btConvexHullShape`).